phoenix-core/src/main/java/org/apache/phoenix/execute/BaseQueryPlan.java

```diff
@@ -500,7 +500,10 @@ public ExplainPlan getExplainPlan() throws SQLException {
         if (context.getScanRanges() == ScanRanges.NOTHING) {
             return new ExplainPlan(Collections.singletonList("DEGENERATE SCAN OVER " + getTableRef().getTable().getName().getString()));
         }
-        
+
+        // Initialize dummy iterator to get the stats.
```

Comment:
Comment that you need the stats prior to optimization as they help drive cost based decisions.